### PR TITLE
* better path management for internal occam applications w/ submods o…

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
…r better if not

Related to og dev.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -17,8 +17,8 @@ $dimage = ENV.fetch("DIMAGE", "netbricks")
 $dtag = ENV.fetch("DTAG", "latest")
 $dproject = ENV.fetch("DPROJECT", "williamofockham")
 $ogpath = ENV.fetch("OGPATH", "../../og")
-$nbpath = ENV.fetch("NBPATH", "../netbricks")
-$mgpath = ENV.fetch("MGPATH", "../moongen")
+$nbpath = ENV.fetch("NBPATH", "../NetBricks")
```
Comment (zeeshanlakhani) [inline Vagrantfile]: @clinedome keeping this for default paths (like when not og'ing). OG's `make vm` sets these appropriately (to `netbricks`, etc...).

Comment (clinedome) [reply Vagrantfile]: tbh, might be best to accept either. could just check for `moongen` or `MoonGen` if the path isn't set. Might as well have the script be permissive -- assuming it'll just be overridden by another script is bound to bite us again later.

File: Vagrantfile [hunk 2]
```diff
@@ -28,18 +28,22 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
...
-    config.vm.synced_folder $nbpath, "/netbricks", disabled: false
-  end
-  if path_exists?($mgpath)
-    config.vm.synced_folder $mgpath, "/moongen", disabled: false
-  end
   if path_exists?($ogpath)
     config.vm.synced_folder $ogpath, "/og", disabled: false
-  end
+    config.vm.synced_folder ".", "/vagrant", disabled: true
+    config.vm.provision "shell", inline: "echo 'cd /og' >> /home/vagrant/.bashrc", run: "always"
```
Comment (clinedome) [inline Vagrantfile]: I wonder if we should drop all `og` mentions out of this file. We could have essentially an `$EXTRA_MOUNT` and `$EXTRA_MOUNT_POINT` which could be set by the makefile. I know there's some logic around syncing that relies on the existence of the `$OGPATH`, but we could probably work that out. There's not a ton of reference to `og`, I wonder if we can make this file work for both cases without being explicit about the `og` stuff.

Comment (zeeshanlakhani) [reply Vagrantfile]: yeah, ok, i can look into it. sure.

File: Vagrantfile (hunk 1 above)
Comment (zeeshanlakhani) [reply Vagrantfile]: I think we can handle this better later, as the repo names are normally MoonGen and NetBricks :).

**Pull request merged**